TNL-6118: Replaced Tire with elasticsearch-model (new)

robrap opened the pull request:
This is the same as https://github.com/edx/cs_comments_service/pull/220, except the fix for the old rake search:rebuild task was added as a separate commit at the start, because it may be needed by OpenCraft.

We should continue the review off this PR.

FYI: @clintonb @dianakhuang @maxrothman

clintonb commented on README.rst:
```diff
@@ -30,15 +29,63 @@ This service relies on Elasticsearch and MongoDB. By default the service will us
+    $ bin/rake search:initialize
+
+To rebuild a new index from the database and then point the alias 'content' to it, you can use the
+rebuid_index task. This task will also run catchup before and after the alias is moved, to minimize time where the alias
```
s/rebuid_index/rebuild_index

robrap commented:
@dianakhuang @clintonb: 
1. I found some minor issues with the rake tasks that you can see in "Fix rebuild_index and catchup."
2. I added some rake tests, because I wasn't happy with all the last minute failures.  Unfortunately, the tests didn't catch the issues that I fixed in "Load app.rb from Rakefile.", when I tested without this fix.
3. The test didn't cover the batch_size or sleep_time parameters, so I'm not sure yet if it would have caught errors I fixed in "Fix rebuild_index and catchup.".

I need to determine:
1. If it makes sense to try to load a different spec_helper that tries to match what is happening in rake for rake tests, or if that could just lead to the same issue?  This feels complicated.
2. Do I leave out the rake tests altogether?  Do I leave them in, but stub out the helper so it just tests the shell?

I just need to talk this over with someone for a moment to decide how to proceed.
